### ISSUE TITLE
`draw_wcsaxes` broken on matplotlib dev

### ISSUE THREAD

**Issue by dhomeier** [astropy/astropy#14174]
### Description
Tests agains `matplotlib>=3.7.0dev` are failing on
`TypeError: TickLabels.draw() got an unexpected keyword argument 'bboxes'`
e.g. in this CI run for [test-devdeps](https://github.com/astropy/astropy/actions/runs/3663728707/jobs/6193696316#step:9:1822).

### Steps to Reproduce
With mpl dev version installed
```python
import astropy
astropy.test(package='visualization.wcsaxes')
```
### System Details
Platform: Linux-5.15.0-1024-azure-x86_64-with-glibc2.35

Executable: /home/runner/work/astropy/astropy/.tox/py311-test-devdeps/bin/python
Full Python Version: 
3.11.0 (main, Oct 26 2022, 10:25:18) [GCC 11.2.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.25.0.dev0+165.g6f9237e91
Scipy: 1.11.0.dev0+1141.46811be
Matplotlib: 3.7.0.dev942+gcca039da6b
PyERFA: 2.0.0.2.dev13+gfb1f520
asdf: 2.14.3.dev14+gcc51e29

**Comment by dstansby** [comment]
This is very odd, I can't reproduce it locally (on currently MPL main and astropy main), and the error (`TypeError: TickLabels.draw() got an unexpected keyword argument 'bboxes'`) suggests it's a issue internal to `astropy`, as `TickLabels` has its own `draw()` method.

**Comment by dstansby** [comment]
Debugging, and for some reason `self.ticklabels` is a `maptlotlib.text.Text` object, instead of being a `astropy.visualization.wcsaxes.ticklabels.TickLabels` object. That explains the error, next step is to work out why this has happened.

As an aside, I suspect this issue is another reason to make https://github.com/astropy/astropy/issues/12672 happen

**Comment by dstansby** [comment]
Reproduced locally, and the error bisects to https://github.com/matplotlib/matplotlib/commit/e83ff55bb35b2959761d494b17effbcf6f5cc45e

**Comment by dstansby** [comment]
Commented at https://github.com/matplotlib/matplotlib/pull/24484#issuecomment-1349501412 with the issue, will see what Matplotlib says.

**Comment by dhomeier** [comment]
Thanks!

**Comment by dhomeier** [comment]
Fixed in matplotlib/matplotlib#24720, thanks all!
